source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -599,6 +599,8 @@ Network::FilterStatus Filter::onNewConnection() {
   return establishUpstreamConnection();
 }
 
+bool Filter::startUpstreamSecureTransport() { return upstream_->startUpstreamSecureTransport(); }
```

Comment:
Emm... seems like there should be an assertion to ensure upstream_ isn't nullptr.